src/test/java/software/amazon/cloudwatchlogs/emf/model/MetricDirectiveThreadSafetyTest.java

```diff
@@ -0,0 +1,75 @@
+package software.amazon.cloudwatchlogs.emf.model;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import org.junit.After;
+import org.junit.Test;
+
+public class MetricDirectiveThreadSafetyTest {
+    private volatile Throwable throwable = null;
+
+    @Test
+    public void testConcurrentPutMetricWithDifferentKey() throws InterruptedException {
+        MetricDirective metricDirective = new MetricDirective();
+        Thread[] threads = new Thread[100];
+        for (int i = 0; i < 100; i++) {
+            final int id = i;
+            threads[i] =
+                    new Thread(
+                            () -> {
+                                try {
+                                    metricDirective.putMetric("Metric-" + id, id);
```

Comment:
let's make each thread run longer to assure more concurrency. Currently I don't know how many threads can actually execute at same time.
one way we can do is introduce thread.sleep 
eg

```
long targetTimestampToRun = System.currentTimeMillis() + 10s; // all threads should target running on this timestamp

for (int i = 0; i < 100; i++) {
    thread[i] = new Thread( () -> {
                       Thread.Sleep(targetTimestampToRun - System.currentTimeMillis());  // try to make all threads run at same time
                        for (j=0; j<1000; j++) metricDirective.putMetric() // run 1000 times putMetric to increase the chance of concurrency
                        }
   thread[i].start()
}

//verify result
```
```